Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -0,0 +1,213 @@
+//
+// Copyright 2023 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may obtain a copy
+// of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+
+@testable import AEPCore
+import AEPEdge
+import AEPEdgeIdentity
+import AEPServices
+import Foundation
+import XCTest
+
+class NetworkTestingDelegate: NetworkRequestDelegate {
+    var testCaseCompletion: (HttpConnection) -> ()
+    func handleNetworkResponse(httpConnection: AEPServices.HttpConnection) {
+        print("Delegate received httpConnection: \(httpConnection)")
+        testCaseCompletion(httpConnection)
+    }
+    init() {
+        testCaseCompletion = { httpConnection in
+            
+        }
+    }
+}
+
+/// This test class validates proper intergration with upstream services, specifically Edge Network
+class UpstreamIntegrationTests: XCTestCase {
+    private var edgeEnvironment: EdgeEnvironment = .prod
+    private var edgeLocationHint: EdgeLocationHint? = nil
+    
+    private let testingDelegate = NetworkTestingDelegate()
+    
+    let LOG_SOURCE = "SampleFunctionalTests"
+    
+    let asyncTimeout: TimeInterval = 10
+
+    override func setUp() {
+        let networkService = IntegrationTestNetworkService()
+        networkService.testingDelegate = testingDelegate
+        ServiceProvider.shared.networkService = networkService
+        
+        // Extract Edge Network environment level from shell environment
+        if let environment = EdgeEnvironment() {
+            self.edgeEnvironment = environment
+        }
+        print("Using Edge Network environment: \(edgeEnvironment.rawValue)")
+        // Extract Edge location hint from shell environment
+        if let locationHint = EdgeLocationHint() {
+            self.edgeLocationHint = locationHint
+        }
```

Comment:
L50-57 + L68-76 
we should have a helper method that sets up the config based on the env params